.github/workflows/create-release-branch.yml

```diff
@@ -0,0 +1,87 @@
+name: create-release-branch
+on:
+  workflow_dispatch:
+    inputs:
+      # When the workflow is executed manually, the user can select whether the branch should correspond to a major,
+      # minor, or patch release.
+      release-type:
+        type: choice
+        description: what kind of release?
+        options:
+          - major
+          - minor
+          - patch
+        required: true
+  schedule:
+    # Cron syntax is "minute[0-59] hour[0-23] date[1-31] month[1-12] day[0-6]". '*' is 'any value,' and multiple values
+    # can be specified with comma-separated lists. All times are UTC.
+    # So this expression means "run at 12 PM UTC, every Friday".
+    - cron: "0 12 * * 5"
+
+jobs:
+  create-release-branch:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    outputs:
+      branch-name: ${{ steps.create-branch.branch_name }}
+    steps:
+      # Depending on circumstances, we may want to exit early instead of running the job to completion.
+      - name: Exit early if applicable
+        # This step is skipped if the workflow was invoked manually, since such an invocation should bypass validations
+        # surrounding what day it is.
+        if: $GITHUB_EVENT_NAME != 'workflow_dispatch'
+        run: |
+          # `date -u` returns UTC datetime, and `%u` formats the output to be the day of the week, with 1 being Monday,
+          # 2 being Tuesday, etc.
+          TODAY_DOW=$(date -u +%u)
+          # Exit early if it's not Friday.
+          [[ $TODAY_DOW != 5 ]] && echo "Not a Friday, should not run" && exit 1
+          # This `date` expression returns the last Tuesday of the month, which is our Release Day. %d formats the output
+          # as the day of the month (1-31).
+          NEXT_RELEASE_DATE=$(date -u -v1d -v+1m -v-1d -v-tue +%d)
+          # This `date` expression returns next Tuesday, and `%d` formats the output as the day of the month (1-31).
+          UPCOMING_TUESDAY_DATE=$(date -u -v+tue +%d)
+          # If next Tuesday isn't Release Day, exit early.
+          [[ $NEXT_RELEASE_DATE != $UPCOMING_TUESDAY_DATE ]] && echo "Wrong Friday" && exit 1
```

Comment:
Very nice. I didn't realize the power of the -v adjust flag until now. Cool. My only thought here is that instead of saying "Wrong Friday" with exit 1 (which makes things error). Should we exit early without an error? Maybe say "Quit early since today isn't the Friday before the last Tuesday of the month" and then just stop the workflow.

One thing you can do to make this happen is to move the cron schedule and this conditional code to another workflow called create-release-branch-from-cron or something which then triggers create-release-branch only if it is the Friday before the last Tuesday of the month. Basically, have a workflow just for the cron portion. This would avoid exit 1 error notifcations showing up in our github actions page.